### PR TITLE
50942 - get readme file from fss largemediaexchangeset

### PR DESCRIPTION
**Pull request by shailen14599**
Code changes and Unit tests
Functional Test
Temporary code in FulfilmentDataSevice (will remove while download enc branch changes)

### CODE REVIEW

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/IFulfilmentDataService.cs [hunk 1]
```diff
@@ -7,5 +7,6 @@ public interface IFulfilmentDataService
     {
         Task<string> CreateExchangeSet(SalesCatalogueServiceResponseQueueMessage message, string currentUtcDate);
         Task<string> CreateLargeExchangeSet(SalesCatalogueServiceResponseQueueMessage message, string currentUtcDate, string largeExchangeSetFolderName);
+        Task GetReadmeFiles(string batchId, string exchangeSetPath, string correlationId);
```
Comment (harshal11869) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/IFulfilmentDataService.cs:10]: No need to add this method in Interface, please remove it.

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentDataService.cs [hunk 2]
```diff
@@ -293,5 +308,25 @@ public async Task CreatePosFolderStructure(string largeMediaExchangeSetPath)
             fileSystemHelper.CheckAndCreateFolder(largeMediaExchangeSetAdcPath);
             await Task.CompletedTask;
         }
+
+        public async Task GetReadmeFiles(string batchId, string exchangeSetPath, string correlationId)
```
Comment (harshal11869) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentDataService.cs:312]: make this method private as not used anywhere else

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentDataService.cs [hunk 3]
```diff
@@ -293,5 +308,25 @@ public async Task CreatePosFolderStructure(string largeMediaExchangeSetPath)
...
+            {
+                var encFolder = Path.Combine(directory.ToString(), fileShareServiceConfig.Value.EncRoot);
+                encFolderList.Add(encFolder);
+            }
+            List<Task> ParallelCreateFolderTasks = new List<Task> { };
+
+            Parallel.ForEach(encFolderList, encFolderList =>
+            {
+                ParallelCreateFolderTasks.Add(DownloadReadMeFile(batchId, encFolderList.ToString(), correlationId));
+            });
```
Comment (harshal11869) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentDataService.cs:327]:
```suggestion
            Parallel.ForEach(encFolderList, encFolder =>
            {
                ParallelCreateFolderTasks.Add(DownloadReadMeFile(batchId, encFolder, correlationId));
            });
```

**Review by harshal11869 — CHANGES_REQUESTED** [review]

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/IFulfilmentDataService.cs (hunk 1 above)
Comment (abhijeet14992) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/IFulfilmentDataService.cs]: Resolved

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentDataService.cs (hunk 3 above)
Comment (abhijeet14992) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentDataService.cs]: Resolved as suggested name changes

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentDataService.cs [hunk 4]
```diff
@@ -293,5 +308,26 @@ public async Task CreatePosFolderStructure(string largeMediaExchangeSetPath)
             fileSystemHelper.CheckAndCreateFolder(largeMediaExchangeSetAdcPath);
             await Task.CompletedTask;
         }
+
+        private async Task GetReadmeFiles(string batchId, string exchangeSetPath, string correlationId)
```
Comment (harshal11869) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentDataService.cs:312]: Change the method name please
```suggestion
        private async Task DownloadReadMeFile(string batchId, string exchangeSetPath, string correlationId)
```

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentDataService.cs [hunk 5]
```diff
@@ -119,13 +119,28 @@ public async Task<string> CreateExchangeSet(SalesCatalogueServiceResponseQueueMe
...
+            #endregion
+
             List<Task> ParallelCreateFolderTasks = new List<Task> { };
             var dvdNumbers = Enumerable.Range(1, 2);
 
             Parallel.ForEach(dvdNumbers, dvdNumber =>
             {
                 ParallelCreateFolderTasks.Add(CreatePosFolderStructure(Path.Combine(homeDirectoryPath, currentUtcDate, message.BatchId, String.Format(largeExchangeSetFolderName, dvdNumber.ToString()))));
                 ParallelCreateFolderTasks.Add(fulfilmentAncillaryFiles.CreateMediaFile(message.BatchId, Path.Combine(homeDirectoryPath, currentUtcDate, message.BatchId, String.Format(largeExchangeSetFolderName, dvdNumber.ToString())), message.CorrelationId, dvdNumber.ToString()));
+                ParallelCreateFolderTasks.Add(GetReadmeFiles(message.BatchId, Path.Combine(homeDirectoryPath, currentUtcDate, message.BatchId, String.Format(largeExchangeSetFolderName, dvdNumber.ToString())), message.CorrelationId));
```
Comment (harshal11869) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentDataService.cs:143]:
```suggestion
                ParallelCreateFolderTasks.Add(DownloadReadMeFile(message.BatchId, Path.Combine(homeDirectoryPath, currentUtcDate, message.BatchId, String.Format(largeExchangeSetFolderName, dvdNumber.ToString())), message.CorrelationId));
```

**Review by harshal11869 — CHANGES_REQUESTED** [review]

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentDataService.cs (hunk 4 above)
Comment (abhijeet14992) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentDataService.cs]: Changed the method name as suggested

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentDataService.cs (hunk 5 above)
Comment (abhijeet14992) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentDataService.cs]: Resolved

**Review by ajasonca — APPROVED** [review]
Looks good

**Pull request merged**